franklin/core/helpers.py

```diff
@@ -104,7 +104,11 @@ def authenticate(self, request):
             backend = 'github'
             backend = load_backend(strategy, backend,
                                    reverse(path, args=(backend,)))
-            user = backend.do_auth(access_token=oauth_token)
+            try:
+                user = backend.do_auth(access_token=oauth_token)
+            except requests.HTTPError as e:
+                msg = e.response.json()
+                raise exceptions.AuthenticationFailed(msg)
```

Comment:
got this error when I was testing @camiriveron github token on my local franklin, which uses a different github application. So yea, I can confirm that you can only auth with tokens for that environments app.